mesonbuild/interpreter/interpreter.py

```diff
@@ -1752,9 +1752,10 @@ def func_vcs_tag(self, node: mparser.BaseNode, args: T.List['TYPE_var'], kwargs:
         vcs_cmd = kwargs['command']
         source_dir = os.path.normpath(os.path.join(self.environment.get_source_dir(), self.subdir))
         if vcs_cmd:
-            maincmd = self.find_program_impl(vcs_cmd[0], required=False)
-            if maincmd.found():
-                vcs_cmd[0] = maincmd
+            if isinstance(vcs_cmd[0], (str, mesonlib.File)):
```

Comment:
This should emit a `FeatureNew.single_use()` if it isn't a string or File.

Also it turns out that we "accidentally" started accepting a File in commit b402817fb6f0392812bfa272bdbc05c9c30139fa (/cc @dcbaker) so that should emit a FeatureNew since 0.62, I guess.